File: cms-user/src/models/CheckOutRequest.ts
export interface ICheckOutRequest {
    addressLine: string;
    paymentMethod?: string;
    cartItemIds?: string[];
}

export class CheckOutRequest {
    public addressLine: string;
    public paymentMethod?: string;
    public cartItemIds?: string[];

    constructor(
        addressLine: string,
        paymentMethod?: string,
        cartItemIds?: string[]
    ) {
        this.addressLine = addressLine.trim();
        this.paymentMethod = paymentMethod;
        this.cartItemIds = cartItemIds;
    }

    toPayload(): ICheckOutRequest {
        return {
            addressLine: this.addressLine,
            paymentMethod: this.paymentMethod,
            cartItemIds: this.cartItemIds
        };
    }
}

File: src/main/java/org/example/websitesalephone/dto/cart/CheckOutRequest.java
package org.example.websitesalephone.dto.cart;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CheckOutRequest {

    private String addressLine;
    private String paymentMethod;
    private List<String> cartItemIds;

}
